gef.py

```diff
@@ -9293,20 +9313,18 @@ def build_line(self, name: str, color: str, address_val: int, got_address: int)
     @parse_arguments({"symbols": [""]}, {"--all": False})
     def do_invoke(self, _: List[str], **kwargs: Any) -> None:
         args : argparse.Namespace = kwargs["arguments"]
-        if args.all:
-            vmmap = gef.memory.maps
-            mapfiles = set(mapfile.path for mapfile in vmmap if
-                           pathlib.Path(mapfile.realpath).is_file() and
-                           mapfile.permission & Permission.EXECUTE)
-            for mapfile in mapfiles:
-                self.print_got_for(mapfile, args.symbols)
-        else:
-            self.print_got_for(str(gef.session.file), args.symbols)
-
-    def print_got_for(self, file: str, argv: List[str]) -> None:
+        vmmap = gef.memory.maps
+        mapfiles = [mapfile for mapfile in vmmap if
+                    (True if args.all else mapfile.path == str(gef.session.file)) and
```

Comment:
`(bool(args.all) or mapfile.path == str(gef.session.file)) and ...`

Reply:
Yeah, I realized later that I'd made that more complex than necessary.